src/ImageIO/CGImageAnimation.cs

```diff
@@ -0,0 +1,141 @@
+//
+// ImageIO - CGImageAnimation.cs
+//
+// Authors:
+//	Whitney Schmidt  <whschm@microsoft.com>
+//
+// Copyright 2020, Microsoft Corp.
+//
+
+using System;
+using System.Runtime.InteropServices;
+using CoreFoundation;
+using CoreGraphics;
+using Foundation;
+using ObjCRuntime;
+
+namespace ImageIO
+{
+
+    public partial class CGImageAnimation
+    {
+
+        public delegate void CGImageSourceAnimationBlock (nint index, CGImage image, out bool stop);
+
+        [Introduced (PlatformName.MacOSX, 10, 15, PlatformArchitecture.All)]
+        [Introduced (PlatformName.iOS, 13, 0, PlatformArchitecture.All)]
+        [DllImport (Constants.ImageIOLibrary)]
+        static extern int CGAnimateImageAtURLWithBlock ( /* CFURLRef */ IntPtr url, /* CFDictionaryRef _iio_Nullable */ IntPtr options, /* CGImageSourceAnimationBlock */ ref BlockLiteral block);
+
+        [Introduced (PlatformName.MacOSX, 10, 15, PlatformArchitecture.All)]
+        [Introduced (PlatformName.iOS, 13, 0, PlatformArchitecture.All)]
+        [DllImport (Constants.ImageIOLibrary)]
+        static extern int CGAnimateImageDataWithBlock ( /* CFDataRef _Nonnull */ IntPtr data, /* CFDictionaryRef _Nullable */ IntPtr options, /* CGImageSourceAnimationBlock _Nonnull */ ref BlockLiteral block);
+
+        [Introduced (PlatformName.MacOSX, 10, 15, PlatformArchitecture.All)]
+        [Introduced (PlatformName.iOS, 13, 0, PlatformArchitecture.All)]
+        [BindingImpl (BindingImplOptions.Optimizable)]
+        public int AnimateImage (NSUrl url, NSDictionary options, [BlockProxy (typeof (NIDCGImageSourceAnimationBlock))] CGImageSourceAnimationBlock block)
+        {
+#if IOS && ARCH_32
+            throw new PlatformNotSupportedException ();
+#else
+            if (url == null)
+                throw new ArgumentNullException (nameof (url));
+            if (block == null)
+                throw new ArgumentNullException (nameof (block));
+
+            int ret;
+            BlockLiteral block_block;
+            block_block = new BlockLiteral ();
+            block_block.SetupBlockUnsafe (SDCGImageSourceAnimationBlock.Handler, block);
+
+            try {
+                ret = CGAnimateImageAtURLWithBlock (url.Handle, options.GetHandle (), ref block_block);
+            } finally {
+                block_block.CleanupBlock ();
+            }
+
+            return ret;
+#endif
+        }
+
+        [Introduced (PlatformName.MacOSX, 10, 15, PlatformArchitecture.All)]
+        [Introduced (PlatformName.iOS, 13, 0, PlatformArchitecture.All)]
+        [BindingImpl (BindingImplOptions.Optimizable)]
+        public int AnimateImage (NSData data, NSDictionary options, [BlockProxy (typeof (NIDCGImageSourceAnimationBlock))] CGImageSourceAnimationBlock block)
```

Comment:
```suggestion
        public int AnimateImage (NSData data, NSDictionary options, [BlockProxy (typeof (NIDCGImageSourceAnimationBlock))] CGImageSourceAnimationHandler handler)
```

Reply:
This returns an `int` which in reality it is an `OSStatus` but I wonder if we can do even better, there is `CGImageAnimationStatus` inside `CGImageAnimation.h` I think it is better to return `CGImageAnimationStatus` 

@spouliot what do you think?